pytorch_lightning/core/lightning.py

```diff
@@ -1605,21 +1609,37 @@ def to_torchscript(
             This LightningModule as a torchscript, regardless of whether file_path is
             defined or not.
         """
-
         mode = self.training
-        with torch.no_grad():
-            if method == 'script':
-                torchscript_module = torch.jit.script(self.eval(), **kwargs)
-            elif method == 'trace':
-                # if no example inputs are provided, try to see if model has example_input_array set
-                if example_inputs is None:
-                    example_inputs = self.example_input_array
-                # automatically send example inputs to the right device and use trace
-                example_inputs = self.transfer_batch_to_device(example_inputs, device=self.device)
-                torchscript_module = torch.jit.trace(func=self.eval(), example_inputs=example_inputs, **kwargs)
-            else:
-                raise ValueError(f"The 'method' parameter only supports 'script' or 'trace', but value given was:"
-                                 f"{method}")
+
+        if method == 'script':
+            torchscript_module = torch.jit.script(self.eval(), **kwargs)
+        elif method == 'trace':
+            # if no example inputs are provided, try to see if model has example_input_array set
+            if example_inputs is None:
+                if self.example_input_array is None:
+                    raise ValueError(
+                        'Choosing method=`trace` requires either `example_inputs`'
+                        ' or `model.example_input_array` to be defined'
+                    )
+                example_inputs = self.example_input_array
+
+            if isinstance(example_inputs, Tensor):
+                example_inputs = (example_inputs,)
+            elif not (
+                isinstance(example_inputs, collections.Sequence)
+                and all(isinstance(inp, Tensor) for inp in example_inputs)
+            ):
+                raise ValueError(
+                    "Could not export to torchscript since example_inputs is neither a Tensor nor tuple of Tensors"
+                )
```

Comment:
why are all these custom error messages necessary? Don't we want to let torch handle the error reporting?
The `to_torchscript` and `to_onnx` methods should be kept simple and just pass the data to their torch functions. 
They can provide a convenience interface but it is expected that user overrides these to provide their own customizations of how and which parts of the model gets exported.

Reply:
pytorch doesn't handle these errors. For eg. in case of `to_onnx`, it you pass a list_inputs like `model(*list_inputs)` manually, it will work but in `torch.onnx.export(model, list_inputs)` it won't and will throw some weird error. Also with other kind of inputs even if they are acceptable by the `model.forward`, it gives weird results in case of torchscript. In the [docs](https://pytorch.org/docs/stable/_modules/torch/onnx.html#export) they just mention the kind of inputs required but there is no error handling if the inputs are not appropriate. So I thought it would be better to do that in lightning.

Reply:
ok sure, in this case I tend to agree. Note that if in a future version pytorch adds support for new input types, we will have to change the logic here which we otherwise wouldn't if pytorch did the error handling

Reply:
Yeah sure, I'll keep an eye on that.